[BUG] - removeByCondition get "slice bounds out of range"

### Describe the bug

Loop in `removeByCondition` (scheduler.go) will keep iterating by the original value because the range loop copies the values from the slice to a local variable n; updating n will not affect the slice. It will result in panic because `slice bounds out of range` when the index of iteration goes beyond the current length of `s.jobs`

### To Reproduce

Steps to reproduce the behavior:

1. Run example code
2. Will result in this error: 
```
2020-07-11 08:00:11 +0000 UTC
panic: runtime error: slice bounds out of range [11:10]

goroutine 1 [running]:
github.com/go-co-op/gocron.removeAtIndex(...)
	/home/ridwanakf/go/pkg/mod/github.com/go-co-op/gocron@v0.2.0/scheduler.go:248
github.com/go-co-op/gocron.(*Scheduler).removeByCondition(0xc000032100, 0xc000098da8)
	/home/ridwanakf/go/pkg/mod/github.com/go-co-op/gocron@v0.2.0/scheduler.go:218 +0x23b
github.com/go-co-op/gocron.(*Scheduler).Remove(0xc000032100, 0x4dfda0, 0x515080)
	/home/ridwanakf/go/pkg/mod/github.com/go-co-op/gocron@v0.2.0/scheduler.go:203 +0x66
main.main()
	/home/ridwanakf/go/src/github.com/ridwanakf/personal-go-projects/cron/main.go:78 +0x1340
exit status 2
```

### Version

commit SHA: 4cc1bfb73568aeb8048dac6902e6b43df7d089ae

### Expected behavior

Should delete all expected jobs without resulting in panic

### Additional context

<!---Add any other context about the problem here.--->

--- a/scheduler.go
+++ b/scheduler.go
@@ -213,9 +213,11 @@ func (s *Scheduler) RemoveByReference(j *Job) {
 }
 
 func (s *Scheduler) removeByCondition(shouldRemove func(*Job) bool) {
-	for i, job := range s.jobs {
-		if shouldRemove(job) {
+	for i := 0; i < len(s.jobs); {
+		if shouldRemove(s.jobs[i]) {
 			s.jobs = removeAtIndex(s.jobs, i)
+		} else {
+			i++
 		}
 	}
 }

--- a/scheduler.go
+++ b/scheduler.go
@@ -213,13 +213,13 @@ func (s *Scheduler) RemoveByReference(j *Job) {
 }
 
 func (s *Scheduler) removeByCondition(shouldRemove func(*Job) bool) {
-	for i := 0; i < len(s.jobs); {
-		if shouldRemove(s.jobs[i]) {
-			s.jobs = removeAtIndex(s.jobs, i)
-		} else {
-			i++
+	retainedJobs := make([]*Job, 0)
+	for _, job := range s.jobs {
+		if !shouldRemove(job) {
+			retainedJobs = append(retainedJobs, job)
 		}
 	}
+	s.jobs = retainedJobs
 }
 
 // RemoveJobByTag will Remove Jobs by Tag

--- a/scheduler_test.go
+++ b/scheduler_test.go
@@ -369,11 +369,12 @@ func TestRemove(t *testing.T) {
 	scheduler := NewScheduler(time.UTC)
 	scheduler.Every(1).Minute().Do(task)
 	scheduler.Every(1).Minute().Do(taskWithParams, 1, "hello")
+	scheduler.Every(1).Minute().Do(task)
 
-	assert.Equal(t, 2, scheduler.Len(), "Incorrect number of jobs")
+	assert.Equal(t, 3, scheduler.Len(), "Incorrect number of jobs")
 
 	scheduler.Remove(task)
-	assert.Equal(t, 1, scheduler.Len(), "Incorrect number of jobs after removing 1 job")
+	assert.Equal(t, 1, scheduler.Len(), "Incorrect number of jobs after removing 2 job")
 
 	scheduler.Remove(task)
 	assert.Equal(t, 1, scheduler.Len(), "Incorrect number of jobs after removing non-existent job")

